source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -64,33 +133,28 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
 
         if (materialXDocument.length() == 0)
         {
-            throw MaterialX::Exception("MaterialX document file path is empty.");
+            throw mx::Exception("MaterialX document file path is empty.");
         }
 
-        MString elementPath;
 	    if (parser.isFlagSet(kElementFlag))
 	    {
 		    argData.getFlagArgument(kElementFlag, 0, elementPath);
 	    }
 
-        if (elementPath.length() == 0)
-        {
-            throw MaterialX::Exception("MaterialX element path is empty.");
-        }
-
         MString ogsXmlPath;
         if (parser.isFlagSet(kOgsXmlFlag))
         {
             argData.getFlagArgument(kOgsXmlFlag, 0, ogsXmlPath);
         }
 
         std::unique_ptr<MaterialXData> materialXData{
-            new MaterialXData(materialXDocument.asChar(), elementPath.asChar())
+            new MaterialXData(materialXDocument.asChar(), elementPath.asChar(), Plugin::instance().getLibrarySearchPath())
         };
 
-        if (!materialXData->isValidOutput())
+        elementPath.set(materialXData->getElementPath().c_str());
```

Comment:
I've made it so that if you don't set an element path (we scan for the first renderable element to use. We do this for our viewers for instance to at least have a starting element. 
This avoid having to scane the text document text for an element which requires knowledge of MTLX and can be a pain if the file is long (see the substance mtlx file :), 

We do a check for valid path here now vs failing if none provided in the command line.